api/routes/generate.py

```
@@ -164,17 +205,26 @@ async def generate_code(request: CodeGenerationRequest) -> CodeGenerationRespons
 
 Modified code:"""
 
-        # Generate modified code using the provider
+        # Generate modified code using the provider with provider-specific API key
+        try:
+            api_key = get_api_key_for_provider(request.provider)
+        except ValueError as e:
+            raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)